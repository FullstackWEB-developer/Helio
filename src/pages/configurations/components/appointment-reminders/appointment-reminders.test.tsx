import i18n from '../../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {act, render} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import AppointmentReminders from './appointment-reminders';
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
   }));
describe("AppointmentReminders tests", () => {
    let container: HTMLDivElement | null;
    const mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            }
        },
        lookupsState: {

        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders AppointmentReminders correctly", async () => {
        let fragment: any;
        await act(async () => {
            const {asFragment} = render(<TestWrapper mockState={mockState}>
                <AppointmentReminders/>          
            </TestWrapper>);
            fragment = asFragment;
        });
        expect(fragment()).toMatchSnapshot();
    });
})
