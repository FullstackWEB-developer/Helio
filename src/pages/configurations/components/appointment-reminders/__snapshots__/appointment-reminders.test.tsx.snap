// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentReminders tests renders AppointmentReminders correctly 1`] = `
<DocumentFragment>
  <div
    class="w-10/12 overflow-auto h-full p-6 pr-4"
  >
    <h5
      class="pb-6"
    >
      configuration.appointment_reminders.title
    </h5>
    <div
      class="flex flex-col space-y-4 items-center justify-center undefined"
    >
      <div>
        <div
          class=""
        >
          <svg
            class="spinner icon-medium "
            viewBox="0 0 40 40"
          >
            <circle
              class="path"
              cx="20"
              cy="20"
              fill="none"
              r="16"
              stroke-width="4"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
